client/src/components/SearchBar.js

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+const SearchBar = ({ videoData, setVideoData }) => {
+  const handleSearch = (e) => {
+    const searchVideoData = videoData.filter((video) =>
+      video.title.toLowerCase().includes(e.target.value.toLowerCase())
+    );
+    setVideoData(searchVideoData);
+  };
```

Comment:
this works really well, I like how it filters as you type and removes the need to click a button! To improve it might be nice to set the video data back to the full list when the search input returns to an empty string